fem/pgridfunc.cpp

```diff
@@ -1059,12 +1054,24 @@ GridFunction ParGridFunction::GetSerialGridFunction(int save_rank,
       if (n_send_recv)
       {
          MPI_Send(&vals[0], n_send_recv, MPITypeMap<real_t>::mpi_type, save_rank, 449,
-                  my_comm);
+                  comm);
       }
    }
 
-   MPI_Barrier(my_comm);
-   return gf_serial;
+   return serial_gf;
+}
+
+GridFunction ParGridFunction::GetSerialGridFunction(int save_rank,
+                                                    Mesh &serial_mesh) const
+{
+   auto *serial_fec = pfes->FEColl()->Clone(pfes->FEColl()->GetOrder());
+   auto *serial_fes = new FiniteElementSpace(&serial_mesh,
```

Comment:
who will be responsible of freeing `serial_fes` ? (looks to me like a leak but most likely I am missing something)

Reply:
If the GridFunction owns the FiniteElementCollection, then it also owns the FiniteElementSpace, and will destroy the space in its destructor:

https://github.com/mfem/mfem/blob/17955e114020af340e9a06a66ebef43e05012d9c/fem/gridfunc.cpp#L156-L164

Reply:
Got it. Many thanks